server/src/server.rs

```diff
@@ -35,14 +39,77 @@ use std::str::FromStr;
 
 async fn handle_put_messages<T: SignalDatabase>(
     state: SignalServerState<T>,
-    address: ProtocolAddress,
+    authenticated_device: AuthenticatedDevice,
+    destination_identifier: ServiceId,
     payload: SignalMessages,
-) -> Result<(), ApiError> {
-    println!("Received message");
-    /* TODO: handle_put_message
-       this depends on ToEnvelope trait being implemented for SignalMessage which depends on Account
-    */
-    todo!()
+) -> Result<SendMessageResponse, ApiError> {
+    if destination_identifier == authenticated_device.account().pni() {
+        return Err(ApiError {
+            status_code: StatusCode::FORBIDDEN,
+            message: "".to_owned(),
+        });
+    }
+
+    let is_sync_message = destination_identifier == authenticated_device.account().aci();
```

Comment:
This is implicitly true when you get here

Reply:
No, because destination_identifier this only checks if it is your own ACI, the destination_identifier can be someone elses id